### ISSUE TITLE
'AssertionError: Neither id, path nor source given' when run inside package

### ISSUE THREAD

**Issue by kirbyfan64** [python/mypy#3552]
Simple reproduction:

```
ryan@DevPC-LX ~/langtest/mypy/multi/c $ touch __init__.py
ryan@DevPC-LX ~/langtest/mypy/multi/c $ touch a.py
ryan@DevPC-LX ~/langtest/mypy/multi/c $ echo 'from .a import *' > b.py
ryan@DevPC-LX ~/langtest/mypy/multi/c $ mypy b.py
Traceback (most recent call last):
  File "/home/ryan/stuff/anaconda/bin/mypy", line 6, in <module>
    exec(compile(open(__file__).read(), __file__, 'exec'))
  File "/media/ryan/stuff/mypy/scripts/mypy", line 6, in <module>
    main(__file__)
  File "/media/ryan/stuff/mypy/mypy/main.py", line 50, in main
    res = type_check_only(sources, bin_dir, options)
  File "/media/ryan/stuff/mypy/mypy/main.py", line 97, in type_check_only
    options=options)
  File "/media/ryan/stuff/mypy/mypy/build.py", line 191, in build
    graph = dispatch(sources, manager)
  File "/media/ryan/stuff/mypy/mypy/build.py", line 1697, in dispatch
    graph = load_graph(sources, manager)
  File "/media/ryan/stuff/mypy/mypy/build.py", line 1806, in load_graph
    ancestor_for=st)
  File "/media/ryan/stuff/mypy/mypy/build.py", line 1275, in __init__
    assert id or path or source is not None, "Neither id, path nor source given"
AssertionError: Neither id, path nor source given
ryan@DevPC-LX ~/langtest/mypy/multi/c $ 
```

This seems kinda bad to me...

**Comment by gvanrossum** [comment]
Actually this is a duplicate of #2974.

**Comment by ethanhs** [comment]
Closed as a duplicate of #2974
